app/services/partner_routing_service.rb

```diff
@@ -62,6 +65,26 @@ def vita_partner_from_source_param
     end
   end
 
+  def vita_partner_from_itin_enabled
+    return unless @intake && @intake.primary_tin_type == "itin" && @intake.primary_ssn.present?
+
+    state = ZipCodes.details(@zip_code)[:state]
+    # TODO: Can people be assigned to Coalitions too? Or only organizations?
+    # coalitions_ids_in_state = StateRoutingTarget.preload(:target).where(state_abbreviation: state, target_type: "Coalition").pluck(:target_id)
+    # do I need to look for these vita partners' child sites and
+    vita_partners_ids_in_state = StateRoutingTarget.preload(:target).where(state_abbreviation: state, target_type: "VitaPartner").pluck(:target_id)
+
+    vita_partners_in_state_itin_enabled = VitaPartner.where(id: vita_partners_ids_in_state, accepts_itin_applicants: true)
+
+    if vita_partners_in_state_itin_enabled.present?
+      @routing_method = :itin_enabled
+      return vita_partners_in_state_itin_enabled.order(Arel.sql('RANDOM()')).first
+    elsif vita_partners_itin_enabled = VitaPartner.where(accepts_itin_applicants: true)
+      @routing_method = :itin_enabled
+      return vita_partners_itin_enabled.order(Arel.sql('RANDOM()')).first if vita_partners_itin_enabled.present?
```

Comment:
Do we also need to check if the vita partner active? But then why does only `from_previous_year_partner` do this check, should we also put it everywhere else?